### PR TITLE
Try all possible Android SDK locations before giving up

### LINKED ISSUE #8618: When searching for Android SDK, look for adb in all PATH elements
Right now, we use `which adb`, but if the user has an `adb` in `/usr/local/bin`, we'll find that and determine that `/usr/` isn't a valid Android SDK.

Instead of using `which adb`, we should look for `adb` in every element of the user's `PATH`

### PR DESCRIPTION
**Pull request by johnmccutchan**
Fixes #8618 

@tvolkert

### CODE REVIEW

File: packages/flutter_tools/lib/src/base/os.dart [hunk 1]
```diff
@@ -34,6 +34,28 @@ abstract class OperatingSystemUtils {
...
   File which(String execName);
 
+  /// Return a list of all paths to `execName` found on the system. Uses the
+  /// PATH environment variable.
+  List<File> whichAll(String execName) {
+    final List<File> result = <File>[];
+    final String pathEnvironmentVariable = platform.environment['PATH'];
+    for (final String pathComponent in _splitPath(pathEnvironmentVariable)) {
+      final String pathToExecName = path.join(pathComponent, execName);
+      if (fs.isFileSync(pathToExecName)) {
```
Comment (goderbauer) [inline packages/flutter_tools/lib/src/base/os.dart]: Should this be `processManager.canRun`? canRun takes care of the fact that executables might have various extensions (exe, bat, etc) on Windows.

File: packages/flutter_tools/lib/src/base/os.dart [hunk 2]
```diff
@@ -5,7 +5,7 @@
 import 'dart:async';
 
 import 'package:archive/archive.dart';
-
+import 'package:pat/path.dart' as path;
```
Comment (goderbauer) [inline packages/flutter_tools/lib/src/base/os.dart]: Instead of importing `path`, use `fs.path`.

Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/base/os.dart]: Done.

File: packages/flutter_tools/lib/src/base/os.dart (hunk 1 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/base/os.dart]: Done.

**Comment by tvolkert** [comment]
LGTM, but tests aren't happy because they [mock out](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/src/context.dart#L154) `OperatingSystemUtils`, and thus are getting `null` when `whichAll()` is called.

**Pull request merged**